aya/src/programs/tc.rs

```diff
@@ -228,6 +228,92 @@ define_link_wrapper!(
     TcLinkId
 );
 
+/// [`SchedClassifierLink`] is the link type used by [`SchedClassifier`] programs.
+///
+/// # Examples
+///
+/// ```no_run
+/// # #[derive(Debug, thiserror::Error)]
+/// # enum Error {
+/// #     #[error(transparent)]
+/// #     IO(#[from] std::io::Error),
+/// #     #[error(transparent)]
+/// #     Map(#[from] aya::maps::MapError),
+/// #     #[error(transparent)]
+/// #     Program(#[from] aya::programs::ProgramError),
+/// #     #[error(transparent)]
+/// #     Bpf(#[from] aya::BpfError)
+/// # }
+/// # let mut bpf = aya::Bpf::load(&[])?;
+/// # tc::qdisc_add_clsact("eth0")?;
+/// # let prog: &mut SchedClassifier = bpf.program_mut("redirect_ingress").unwrap().try_into()?;
+/// # prog.load()?;
+/// #
+/// # use aya::programs::tc::{SchedClassifierLink};
+/// # use aya::programs::{tc, Link, SchedClassifier, TcAttachType};
+/// # use std::mem;
+/// let tc_link_id = prog.attach("eth0", TcAttachType::Ingress)?;
+///
+/// let tc_link = prog.take_link(tc_link_id)?;
+///
+/// // The priority and handle assigned during attach can be accessed as follows:
+/// let priority = tc_link.priority();
+/// let handle = tc_link.handle();
+///
+/// // The following call "forgets" about tc_link so that it is not detached on drop.
+///  mem::forget(tc_link);
+///
+/// // The link can be re-instantiated and detached as shown below.
+/// let new_tc_link = SchedClassifierLink::new_tc_link(
+///     "eth0",
+///     TcAttachType::Ingress,
+///     priority,
+///     handle,
+/// )?;
+/// new_tc_link.detach()?;
+///
+/// # Ok::<(), Error>(())
+/// ```
+impl SchedClassifierLink {
+    /// Creates a new link from the provided values.
+    ///
+    /// This API is used to construct a [`SchedClassifierLink`] where the `if_name`, `attach_type`,
+    /// `priority` and `handle` are already known. This may have been found from a link created by
+    /// [SchedClassifier::attach], the output of the `tc filter` command or from the output of
+    /// another BPF loader.
+    ///
+    /// Note: If you create a link for a program that you do not own, detaching it may have
+    /// unintended consequences.
+    ///
+    /// # Errors
+    /// - If a program is not attached with the provided parameters, calls to
+    ///   [`SchedClassifierLink::detach`] will return a [`TcError::NetlinkError`]
+    pub fn new_tc_link(
```

Comment:
This should just be called `new` I think?

Reply:
Renamed to 'attached' per other comment.